R/cast.R

```diff
@@ -160,22 +160,14 @@ vec_coercible_cast <- function(x, to, ..., x_arg = "", to_arg = "") {
 #' @inheritParams vec_cast
 #' @export
 vec_default_cast <- function(x, to, x_arg = "", to_arg = "") {
-  if (is_bare_list(to)) {
-    return(vec_cast_list_default(x, to, x_arg = x_arg, to_arg = to_arg))
-  }
-
   if (is_asis(x)) {
     return(vec_cast_from_asis(x, to, x_arg = x_arg, to_arg = to_arg))
   }
   if (is_asis(to)) {
     return(vec_cast_to_asis(x, to, x_arg = x_arg, to_arg = to_arg))
   }
 
-  if (inherits_only(to, c("vctrs_list_of", "vctrs_vctr"))) {
-    return(vec_cast_to_list_of(x, to, x_arg = x_arg, to_arg = to_arg))
-  }
-
-  if (inherits(to, "vctrs_vctr")) {
+  if (inherits(to, "vctrs_vctr") && !inherits(to, "vctrs_rcrd")) {
```

Comment:
So `vctrs_list_of` goes through `vctr_cast()`?

Reply:
yeah but don't ask me too much about what this means 😄 

I think we need to review the convenience implementations for `vctrs_vctr`. Opened issue #968.